jobsdb/jobsdb.go

```diff
@@ -2484,41 +2483,44 @@ func (jd *HandleT) getProcessedJobsDS(ctx context.Context, ds dataSetT, params G
 	}
 
 	tags := statTags{CustomValFilters: params.CustomValFilters, StateFilters: params.StateFilters, ParameterFilters: params.ParameterFilters}
-	queryStat := jd.getTimerStat("processed_ds_time", &tags)
-	queryStat.Start()
-	defer queryStat.End()
+	start := time.Now()
+	defer jd.getTimerStat("processed_ds_time", &tags).Since(start)
 
-	// We don't reset this in case of error for now, as any error in this function causes panic
-	jd.markClearEmptyResult(ds, allWorkspaces, stateFilters, customValFilters, parameterFilters, willTryToSet, nil)
-
-	var stateQuery, customValQuery, limitQuery, sourceQuery string
+	skipCacheResult := params.AfterJobID != nil
```

Comment:
Explain: we don't update the cache if the query contains an `AfterJobID` parameter